rfcs/63-nursery.md

```diff
@@ -0,0 +1,170 @@
+   3. Marginally increases security risks (2FA + only write permissions) reduce this risk substantially.
+3. Likely to increase the scope of the core engine.
+   1. Feature flags, template games and plugin sets help ensure users only compile and ship the features they care about.
+   2. Adding maintainers ensures that the workload does not increase significantly.
+
+## Rationale and alternatives
+
+### Why should we upstream crates?
+
+In other modular ecosystems, it's very common to have critical functionality outside of the core project.
+This is unfortunate because:
+
+1. It's very challenging for beginners to discover these essential tools.
+2. These tools cannot readily be used in official documentation.
+3. These tools may suddenly go unmaintained.
+4. Work may be divided across competitors (including work done by the core).
+5. Integration with the core may be poor, or suddenly break.
+
+By upstreaming essential tools, we can avoid these problems, and gracefully scale the core organization.
+
+### Why do we need a "nursery" stage?
+
+Ecosystem crates are huge, complex endeavours.
+We want to ensure that the appropriate quality bar is met, but still signal early once a clear candidate has been identified.
+
+The nursery stage serves as a distributed PR review process for the entire crate, breaking down the necessary changes into manageable units of work.
+The added attention helps ensure that community stakeholders have a chance to weigh in before the crate itself is upstreamed.
+
+Finally, as Bevy stabilizes, nursery crates can be rapidly iterated on without following Bevy's stability guarantees.
+
+### Why shouldn't we rename the crate to `bevy_nursery_*` for improved discoverability?
+
+This results in significant bureaucratic work (transferring issues and PRs), risks user confusion and adds pointless migration pain.
+The discoverability issue can be tackled via the Discord channel and the prominent label on Bevy Assets.
+
+### Why do we need to grant merge rights to nursery crate maintainers?
+
+If we do not:
+
+1. Maintainers of popular ecosystem crates are reluctant to submit their crates to the nursery due to the reduced ability to iterate on and fix their projects.
+2. Maintainers of the core engine may become overloaded with work as the scope of the engine grows.
+
+The maintainers of nursery crates have shown their ability to identify, create and lead a high-value project: they're natural candidates for additional responsibility in the core engine.
+
+### Why not use GitHub's Code Owners feature to help scope merge rights?
+
+While [bors has code owners integration](https://bors.tech/rfcs/0357-bors-support-for-codeowners.html),
+it's not clear that it's a useful tool over simple social contract.
+
+Setting up (and maintaining) these lists is tedious bureaucracy.
+This approach would also encourage a less-integrated architecture for organizational reasons, rather than technical ones.
```

Comment:
I encourage you to reconsider the [`.github/CODEOWNERS`](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) functionality.

You have noted in this document that:

> 2. Tracking areas of expertise may get onerous.

And here you have decided against using `CODEOWNERS` to define _merge rights_.

However, `CODEOWNERS` is a tool aimed at raising awareness of a change to an area of code that you care about. That sounds like something you _do_ want. The core functionality of `CODEOWNERS` is to automatically add a set of users, teams, and/or orgs that own the code as a reviewer to a PR. The ability to _require approvals_ from codeowners is a feature of branch protection, but you don't have to switch that on.

---

On a separate topic, [Merge Queue](https://github.com/github/roadmap/issues/370) is a new GitHub feature that is in beta and should replace the functionality of `bors` with much better integration with the rest of GitHub. For examples, PR's merged via a merge queue will show as "merged", not "closed", and the UI for Merge Queue will be part of GitHub itself. I encourage you to try it out when it goes GA (or if you're really curious, maybe try out the beta).

Reply:
I'm very curious about the Merge Queues beta and applied as soon as it was announced. Havent heard back / been accepted :)